motionplan/constraint.go

```diff
@@ -130,11 +130,14 @@ func (c *ConstraintHandler) CheckStateConstraintsAcrossSegment(ci *ik.Segment, r
 
 	for i := 0; i <= steps; i++ {
 		interp := float64(i) / float64(steps)
-		interpConfig := referenceframe.InterpolateInputs(ci.StartConfiguration, ci.EndConfiguration, interp)
-		interpC := &ik.State{Frame: ci.Frame, Configuration: interpConfig}
-		err = resolveStatesToPositions(interpC)
-		if err != nil {
-			return false, nil
+		interpC := &ik.State{Frame: ci.Frame}
+		if ci.StartPosition != nil && ci.EndPosition != nil && ci.StartConfiguration == nil && ci.EndConfiguration == nil {
```

Comment:
1. Is there a name for what it means for the inverse kinematic segment to have no start & end position and to have both a start & end configuration? If so could we comment on what this means or turn this check into a function that tests for that property?

2. Is it possible for 3 out of the 4 checks to succeed or would that be symptomatic of a bug?

Reply:
> Is there a name for what it means for the inverse kinematic segment to have no start & end position and to have both a start & end configuration? If so could we comment on what this means or turn this check into a function that tests for that property?

if there is a name for this, I am unsure what is it.
At the beginning of this function we do `resolveSegmentsToPositions(ci)` which uses the information given by the start & end configuration to provide the segment with its starting and ending position.

> Is it possible for 3 out of the 4 checks to succeed or would that be symptomatic of a bug?

I am a little unsure what you mean by this question. My understanding of what you are asking is: is it possible for 3 out of 4 of the items in the conditional above to be true and if so, is this symptomatic of a bug?

If our starting three pieces of information from a segment are non-nil startPosition, non-nil endPosition, non-nil StartConfiguration. Then we do not enter the first conditional and proceed to the else statement. This itself is a bug , we cannot do `referenceframe.InterpolateInputs` and pass in a nil `EndConfiguration`.

the if statement should be changed to

```
if ci.StartPosition != nil && ci.EndPosition != nil && ci.StartConfiguration == nil || ci.EndConfiguration == nil {
```
https://go.dev/play/p/_FfDDJa7PPp